campaigns/windows-10/logo-swap.html

```diff
@@ -0,0 +1,232 @@
+
+          var brandLogo = document.getElementById('brandLogo');
+          if (brandLogo.src !== undefined) {
+              brandLogo.src = newLogo;
+          } else {
+              brandLogo.style.backgroundImage = 'url(' + newLogo + ')';
+          }
+          brandLogo.className += ' switch-{{ snippet_id }}';
+
+           var buttons = snippet.getElementsByClassName('launch-default');
+           var snippet_content_default = document.getElementById('content-default-{{ snippet_id }}');
+           var snippet_content_not_default = document.getElementById('content-not-default-{{ snippet_id }}');
+           var snippet_text_default = document.getElementById('text-default-{{ snippet_id }}');
+           var snippet_text_success = document.getElementById('text-success-{{ snippet_id }}');
+           var snippet_icon = document.getElementById('icon-{{ snippet_id }}');
+
+           var poll_retry = 0;
+           var check_default_timeout = undefined;
+           var is_firefox_default = undefined;
+           var cta_clicked = false;
+
+           isDefault(function yesDefault () {
+               sendMetric('impression-is-default');
+               showDefaultContent();
+               is_firefox_default = true;
+           }, function noDefault () {
+               sendMetric('impression-is-not-default');
+               showNonDefaultContent();
+               is_firefox_default = false;
+           });
+
+           var check_default_timeout = window.setInterval(function () {
+               if (poll_retry > 500) {
+                   clearInterval(check_default_timeout);
+               }
+               poll_retry++;
+               isDefault(function yesDefault () {
+                   //Default has switched while the snippet is opened
+                   if (is_firefox_default !== true) {
+                       showSuccessContent();
+                       is_firefox_default = true;
+                       if (cta_clicked === true) {
+                           sendMetric('conversion-is-default-changed-after-click');
+                       } else {
+                           sendMetric('conversion-is-default-changed-without-click');
+                       }
+                       clearInterval(check_default_timeout);
+                   }
+               }, function noDefault () {
+                   if (is_firefox_default !== false) {
+                       showNonDefaultContent();
+                       is_firefox_default = false;
+                   }
+               });
+           }, 1000);
+
+           function showDefaultContent () {
+               snippet_text_default.style.display = 'block';
+               snippet_text_success.style.display = 'none';
+               snippet_content_default.style.display = 'block';
+               snippet_content_not_default.style.display = 'none';
+               snippet_icon.src = '{{ icon }}';
+
+               for (var i=0; i < buttons.length; i++) {
```

Comment:
nit: micro optimization to cache buttons.length value `var i=0, k=buttons.length; i<k; i++`